streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamTaskTest.java

```diff
@@ -431,12 +407,108 @@ public void flush() {
             }
         };
         final StreamsMetrics streamsMetrics = new MockStreamsMetrics(new Metrics());
-        final StreamTask streamTask = new StreamTask(new TaskId(0, 0), "appId", partitions, topology, consumer,
-            restoreStateConsumer, createConfig(baseDir), streamsMetrics, stateDirectory, new ThreadCache("testCache", 0, streamsMetrics), new MockTime(), recordCollector);
+        final StreamTask streamTask = new StreamTask(taskId00, "appId", partitions, topology, consumer,
+                                                     restoreStateConsumer, createConfig(baseDir), streamsMetrics,
+                                                     stateDirectory, testCache, time, recordCollector);
         streamTask.flushState();
         assertTrue(flushed.get());
     }
 
+    @Test(expected = IllegalStateException.class)
+    public void shouldThrowIllegalStateExceptionIfCurrentNodeIsNotNullWhenPunctuateCalled() throws Exception {
+        ((ProcessorContextImpl) task.processorContext()).setCurrentNode(processor);
+        task.punctuate(processor, 10);
```

Comment:
wrap with `try-catch` instead of `expected` annotation -- more than one line test.